src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py

```diff
@@ -588,7 +588,7 @@ def _ensure_container_insights_dcr_for_monitoring(cmd, subscription_id, cluster_
 
     # extract subscription ID and resource group from workspace_resource_id URL
     parsed = parse_resource_id(workspace_resource_id)
-    workspace_subscription_id, workspace_resource_group = parsed["subscription"], parsed["resource_group"]
+    workspace_subscription_id = parsed["subscription"]

```

Comment:
1. Is the workspace_resource_id guaranteed to be always lowercase?  Or, is parsed case insensitive?  If not, consider converting to lowercase before extracting
2. Nit - update the comment since you are not extracting resource group

Reply:
BTW, why is this above change needed - it seems to not impact the fix you are making?

Reply:
- Updated to lowercase before extraction, you are correct, it can be uppercase too.

- the variable workspace_resource_group was only being used in the dcr_resource_id and would have become an unused varaible. Hence I was removing it.

Please let me know if this looks good to you, and merge it in if it does.
Thanks

Reply:
@NarayanThiru let me know if this looks good to you or not

Reply:
@NarayanThiru  let me know if this looks good to you or not